src/stores/room-list-v3/RoomListStoreV3.ts

```diff
@@ -78,7 +83,93 @@ export class RoomListStoreV3Class extends AsyncStoreWithClient<EmptyObject> {
+                const room = payload.room;
+                this.addRoomAndEmit(room);
+                break;
+            }
+
+            case "MatrixActions.Event.decrypted": {
+                const roomId = payload.event.getRoomId();
+                if (!roomId) return;
+                const room = this.matrixClient.getRoom(roomId);
+                if (!room) {
+                    logger.warn(`Event ${payload.event.getId()} was decrypted in an unknown room ${roomId}`);
+                    return;
+                }
+                this.addRoomAndEmit(room);
+                break;
+            }
+
+            case "MatrixActions.accountData": {
+                if (payload.event_type !== EventType.Direct) return;
+                const dmMap = payload.event.getContent();
+                for (const userId of Object.keys(dmMap)) {
+                    const roomIds = dmMap[userId];
+                    for (const roomId of roomIds) {
+                        const room = this.matrixClient.getRoom(roomId);
+                        if (!room) {
+                            logger.warn(`${roomId} was found in DMs but the room is not in the store`);
+                            continue;
+                        }
+                        this.addRoomAndEmit(room);
+                    }
+                }
+                break;
+            }
+
+            case "MatrixActions.Room.timeline": {
+                // Ignore non-live events (backfill) and notification timeline set events (without a room)
+                if (!payload.isLiveEvent || !payload.isLiveUnfilteredRoomTimelineEvent || !payload.room) return;
+                this.addRoomAndEmit(payload.room);
+                break;
+            }
+
+            case "MatrixActions.Room.myMembership": {
+                const oldMembership = getEffectiveMembership(payload.oldMembership);
+                const newMembership = getEffectiveMembershipTag(payload.room, payload.membership);
+                if (oldMembership !== EffectiveMembership.Join && newMembership === EffectiveMembership.Join) {
+                    // If we're joining an upgraded room, we'll want to make sure we don't proliferate
+                    // the dead room in the list.
+                    const roomState: RoomState = payload.room.currentState;
+                    const predecessor = roomState.findPredecessor(this.msc3946ProcessDynamicPredecessor);
+                    if (predecessor) {
+                        const prevRoom = this.matrixClient?.getRoom(predecessor.roomId);
+                        if (prevRoom) this.roomSkipList.removeRoom(prevRoom);
+                        else logger.warn(`Unable to find predecessor room with id ${predecessor.roomId}`);
+                    }
+                }
+                this.addRoomAndEmit(payload.room);
+                break;
+            }
+        }
+    }
+
+    private addRoomAndEmit(room: Room): void {
```

Comment:
Missing tsdoc

Reply:
Done in https://github.com/element-hq/element-web/pull/29397/commits/3e32e2d632e24fb61a15348bc03a1875fb83c6c0